client/component/src/SurfacePlot.tsx

```diff
@@ -15,46 +15,43 @@ import { TbGridDots } from 'react-icons/tb';
 
 import PlotToolbar from './PlotToolbar';
 import type { IIconType } from './Modal';
-import type { MP_NDArray, SurfacePlotProps } from './AnyPlot';
+import type { NDT, PlotBaseProps } from './AnyPlot';
 
 /**
- *
- * Represents surface data.
- * @interface SurfaceData
- * @member {string} key - The key.
- * @member {MP_NDArray} values - The surface data values.
- * @member {Domain} domain - The surface data domain.
- * @member {string} surface_scale - The surface data scale.
- * @member {ColorMap} colourMap - The surface colour map.
+ * Represent surface data
  */
 interface SurfaceData {
-  /** The key */
+  /** The object key */
   key: string;
-  /** The surface data values */
-  values: MP_NDArray;
-  /** The surface data domain */
+  /** The data values for the surface plot */
+  values: NDT;
+  /** The domain of the surface data */
   domain: Domain;
-  /** The surface data scale */
-  surface_scale: string;
-  /** The surface colour map */
-  colourMap: ColorMap;
+  /** The colour scale type */
+  surfaceScale: ColorScaleType;
+  /** The colour map (optional) */
+  colourMap?: ColorMap;
 }
 
 /**
- *
- * Renders a surface plot.
+ * Props for the `SurfacePlot` component.
+ */
+interface SurfacePlotProps extends PlotBaseProps, SurfaceData {}
```

Comment:
It needs plotConfig and batonProps

Reply:
`TableDisplayProps` also needs `batonProps`, but just extends from `TableData`. Would it be better to extend `SurfacePlot` just from `SurfaceData` and add in `batonprops` and `plotConfig` rather than add selections fields that don't need to be there.

Reply:
Done